syncopy/tests/test_preproc.py

```diff
@@ -206,6 +221,267 @@ def test_but_parallel(self, testcluster=None):
+            plot_spec(ax, spec, c='0.3', label='unfiltered')
+            annotate_foilims(ax, *self.freq_kw['bp'])
+            ax.set_title(f"Twopass FIRWS, order = {kwargs['order']}")
+
+    def test_firws_kwargs(self):
+
+        """
+        Test order and direction parameter
+        """
+
+        for direction in preproc.availableDirections:
+            kwargs = {'direction': direction,
+                      'order': 200}
+            self.test_firws_filter(**kwargs)
+        for order in [-2, 220, 5.6]:
+            kwargs = {'direction': 'twopass',
+                      'order': order}
+
+            if order < 1 and isinstance(order, int):
+                failed = True
+                try:
+                    self.test_firws_filter(**kwargs)
+                    failed = False
+                except SPYValueError as err:
+                    assert "value to be greater" in str(err)
+                assert failed
+            elif not isinstance(order, int):
+                failed = True
+                try:
+                    self.test_firws_filter(**kwargs)
+                    failed = False
+                except SPYValueError as err:
+                    assert "expected int_like" in str(err)
+                assert failed
+            # valid order
+            else:
+                self.test_firws_filter(**kwargs)
+
+    def test_firws_selections(self):
+
+        sel_dicts = helpers.mk_selection_dicts(nTrials=20,
+                                               nChannels=2,
+                                               toi_min=self.time_span[0],
+                                               toi_max=self.time_span[1],
+                                               min_len=3.5)
+        for sd in sel_dicts:
+            print(sd)            
+            self.test_firws_filter(select=sd, order=200)
+
+    def test_firws_polyremoval(self):
+
+        helpers.run_polyremoval_test(self.test_firws_filter)
+
+    def test_firws_cfg(self):
+
+        cfg = get_defaults(ppfunc)
+
+        cfg.filter_class = 'firws'
+        cfg.order = 200
+        cfg.direction = 'twopass'
+        cfg.freq = 30
+        cfg.filter_type = 'hp'
+
+        result = ppfunc(self.data, cfg)
+
+        # check here just for finiteness
+        assert np.all(np.isfinite(result.data))
+
+    @skip_without_acme
+    def test_firws_parallel(self, testcluster=None):
+
+        ppl.ioff()
+        client = dd.Client(testcluster)
+        all_tests = [attr for attr in self.__dir__()
+                     if (inspect.ismethod(getattr(self, attr)) and 'parallel' not in attr)]
+
+        for test_name in all_tests:
+            test_method = getattr(self, test_name)
+            if 'firws_filter' in test_name:
+                # test parallelisation along channels
+                test_method(chan_per_worker=2)
+            else:
+                test_method()
+        client.close()
+        ppl.ion()
+
+    def test_firws_hilbert_rect(self):
```

Comment:
If I'm not mistaken this code is identical to `test_but_rect` with the exception of the very first `ppfunc` invocation. I think if we're committed to DRY we could unify the codes in a single `test_rect` function that loops over `filter_class='but'` and `filter_class='firws'`

Reply:
Yes I was thinking about this, however the default `order` parameter is so different for such cases, that I think for clarity maybe it's ok like this?

Reply:
OK, let's keep them separate.